File: ViperC-Quick.xctemplate/Objective-C/___FILEBASENAMEASIDENTIFIER___ViewControllerTests.m
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//  Copyright © ___YEAR___ ___ORGANIZATIONNAME___. All rights reserved.
//

#import <Quick/Quick.h>
#import <Expecta/Expecta.h>
#import "___VARIABLE_viperModuleName___ViewController.h"
#import "___VARIABLE_viperModuleName___Router.h"

QuickSpecBegin(___VARIABLE_viperModuleName___ViewControllerTests)

describe(@"___VARIABLE_viperModuleName___ViewController Tests", ^{
    describe(@"+[___VARIABLE_viperModuleName___Router createModule]", ^{
        __block ___VARIABLE_viperModuleName___ViewController *view;

        beforeSuite(^{
            view = (___VARIABLE_viperModuleName___ViewController *) [___VARIABLE_viperModuleName___Router createModule];
        });

        context(@"when created by router", ^{
            it(@"should succeed", ^{
                expect(view).notTo.beNil();
                expect(view.presenter).notTo.beNil();
            });
        });
    });
});

QuickSpecEnd

File: ViperC-Quick.xctemplate/Swift/___FILEBASENAMEASIDENTIFIER___ViewControllerTests.swift
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//  Copyright © ___YEAR___ ___ORGANIZATIONNAME___. All rights reserved.
//

import Quick
import Nimble

//@testable import YOUR_MODULE_NAME

class ___VARIABLE_viperModuleName___ViewControllerTests: QuickSpec {

    override func spec() {
        describe("___VARIABLE_viperModuleName___ViewController Tests") {
            var view: ___VARIABLE_viperModuleName___ViewController!

            beforeSuite {
                view = ___VARIABLE_viperModuleName___Router.createModule() as? ___VARIABLE_viperModuleName___ViewController
            }

            context("when created by router", {
                it("should succeed", closure: {
                    expect(view).notTo(beNil())
                    expect(view.presenter).notTo(beNil())
                })
            })
        }
    }

}
